File: UI/app/src/main/java/com/rikkei/training/ui/dialog/MultiChoiceDialog.java
package com.rikkei.training.ui.dialog;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.content.res.TypedArray;
import android.os.Bundle;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.rikkei.training.ui.R;
import com.rikkei.training.ui.base.BaseDialogFragment;

public class MultiChoiceDialog extends BaseDialogFragment {
    public MultiChoiceDialog(Context context) {
        super(context);
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity())
                .setTitle(R.string.pick_toppings)
                .setMultiChoiceItems(R.array.pick_toppings, null, new DialogInterface.OnMultiChoiceClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i, boolean b) {
                        if (b) {
                            TypedArray toppings = getResources().obtainTypedArray(R.array.pick_toppings);
                            String topping = toppings.getString(i);
                            Toast.makeText(getContext(), topping, Toast.LENGTH_SHORT).show();
                        }
                    }
                })
                .setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {

                    }
                })
                .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {

                    }
                });

        return builder.create();
    }
}

File: UIKotlin/app/src/main/java/com/rikkei/training/uikotlin/dialog/MultiChoiceDialog.kt
package com.rikkei.training.uikotlin.dialog

import android.app.AlertDialog
import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.widget.Toast
import com.rikkei.training.uikotlin.R

import com.rikkei.training.uikotlin.base.BaseDialogFragment

class MultiChoiceDialog(context: Context) : BaseDialogFragment(context) {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(activity)
            .setTitle(R.string.pick_toppings)
            .setMultiChoiceItems(
                R.array.pick_toppings, null
            ) { _, i, b ->
                if (b) {
                    val toppings = resources.obtainTypedArray(R.array.pick_toppings)
                    val topping = toppings.getString(i)
                    Toast.makeText(context, topping, Toast.LENGTH_SHORT).show()
                }
            }
            .setNegativeButton(
                R.string.cancel
            ) { _, _ -> }
            .setPositiveButton(
                R.string.ok
            ) { _, _ -> }

        return builder.create()
    }
}
